hal/src/main/native/systemcore/I2C.cpp

```diff
@@ -29,26 +34,116 @@ extern "C" {
 
 void HAL_InitializeI2C(HAL_I2CPort port, int32_t* status) {
   hal::init::CheckInit();
+
+  if (port != 0) {
+    *status = RESOURCE_OUT_OF_RANGE;
+    hal::SetLastErrorIndexOutOfRange(status, "Invalid Index for I2C", 0, 0,
+                                     port);
+    return;
+  }
+
+  std::scoped_lock lock(digitalI2COnBoardMutex);
+  i2COnboardObjCount++;
+  if (i2COnboardObjCount > 1) {
+    return;
```

Comment:
Should set status code.

Reply:
This one isn't an error. You can have multiple instances. And we don't generally set status codes on success.